chain/stmgr/searchwait.go

```diff
@@ -145,7 +146,20 @@ func (sm *StateManager) SearchForMessage(ctx context.Context, head *types.TipSet
 		return head, r, foundMsg, nil
 	}
 
-	fts, r, foundMsg, err := sm.searchBackForMsg(ctx, head, msg, lookbackLimit, allowReplaced)
+	fts, r, foundMsg, err := sm.searchForIndexedMsg(ctx, mcid, msg)
```

Comment:
As currently written, this will not benefit calls to `WaitForMessage` (the API's `StateWaitMsg`). You can either perform a quick refactor here to speed up that method too, or just open an enhancement issue?

Reply:
let's TODO it for the follow up.

Reply:
added TODO.